tests/link/jax/test_random.py

```diff
@@ -24,7 +24,40 @@ def random_function(*args, **kwargs):
     with pytest.warns(
         UserWarning, match=r"The RandomType SharedVariables \[.+\] will not be used"
     ):
-        return function(*args, **kwargs)
+        return pytensor.function(*args, **kwargs)
+
+
+@pytest.mark.parametrize("random_type", ("generator", "state"))
+def test_rng_io(random_type):
+    """Test explicit (non-shared) input and output RNG types in JAX."""
+    if random_type == "generator":
+        rng = random_generator_type("rng")
+        np_rng = np.random.default_rng(0)
+    else:
+        rng = random_state_type("rng")
+        np_rng = np.random.RandomState(0)
+    jx_rng = jax.random.PRNGKey(0)
+
+    next_rng, x = aer.normal(rng=rng).owner.outputs
+
+    with pytest.warns(
+        UserWarning,
+        match="RandomTypes are implicitly converted to random PRNGKey arrays",
+    ):
+        fn = pytensor.function([rng], [next_rng, x], mode="JAX")
+
+    # Inputs - RNG outputs
+    assert isinstance(fn(np_rng)[0], jax.Array)
+    assert isinstance(fn(jx_rng)[0], jax.Array)
+
+    # Inputs - Value outputs
+    assert fn(np_rng)[1] == fn(np_rng)[1]
+    assert fn(jx_rng)[1] == fn(jx_rng)[1]
+    assert fn(np_rng)[1] != fn(jx_rng)[1]
```

Comment:
Is this the desired behavior, or just how it works out?

Reply:
Just how it works. We are not going out of our way to try and find a 1-1 map between numpy generator and JAX PRNGKeys